### PR TITLE
refactor(checkbox): refactor according the new designs and add outlined

### PR DESCRIPTION
**Pull request by NikosCha**
## Description

Add a new filled property for checkbox and refactor it according to the new designs in here

resolves: #119 

In more details:

* remove uneeded code 
* update hover's shadow color and width/height of checkbox and of parent element
* update shapes and theirs colors according to designs 
* add outline property
* create a new story with props

## Screenshot

Before : 
image

Now: 
image
image
image

## Test Plan

Update the snapshots

### CODE REVIEW

**Comment by panvourtsis** [comment]
can you please change this PR to merge to v2 branch ?

File: src/components/CheckBox/CheckBox.style.ts [hunk 1]
```diff
@@ -22,76 +22,98 @@ export const checkboxWrapperStyle = () => (theme: Theme): SerializedStyles => cs
...
-  position: absolute;
-  opacity: 0; // hide it
-
-  & + label {
-    position: relative;
-    cursor: pointer;
-    padding: 0;
-  }
+const getBackgroundColor = ({ intermediate, checked, filled, theme }: Props & { theme: Theme }) => {
+  let background = '';
```
Comment (panvourtsis) [inline src/components/CheckBox/CheckBox.style.ts:35]: can i suggest remove the `let` and turn all the `background =` to `return {the color required`.

This way will be enforced to split all the if elses to a more understandable way

Comment (panvourtsis) [reply src/components/CheckBox/CheckBox.style.ts:35]: eg

```
if (checked && !intermediate && filled) {
    return `background: ${theme.utils.getColor('lightGray', 700)}`;
  }

  return !filled ? background : `background: ${theme.utils.getColor('lightGray', 400)}`;
```

File: src/components/CheckBox/CheckBox.style.ts [hunk 2]
```diff
@@ -22,76 +22,98 @@ export const checkboxWrapperStyle = () => (theme: Theme): SerializedStyles => cs
...
   }
 
-  // Disabled state label.
-  &:disabled + label {
-    cursor: not-allowed;
+  return background;
+};
+
+const getSymbolColor = ({ intermediate, filled, theme }: Props & { theme: Theme }) => {
+  let color = '';
```
Comment (panvourtsis) [inline src/components/CheckBox/CheckBox.style.ts:56]: also here as well there is no need for defining a `let color` where your default value is always defined in the else operator

e.g
```
if (!filled) {
    return theme.utils.getColor('darkGray', 700);
}
return intermediate ? theme.utils.getColor('lightGray', 700) : 'white';
```

Comment (NikosCha) [reply src/components/CheckBox/CheckBox.style.ts]: done

File: src/components/CheckBox/CheckBox.style.ts (hunk 1 above)
Comment (NikosCha) [reply src/components/CheckBox/CheckBox.style.ts]: done

File: src/components/CheckBox/CheckBox.style.ts [hunk 3]
```diff
@@ -22,76 +22,88 @@ export const checkboxWrapperStyle = () => (theme: Theme): SerializedStyles => cs
...
+};
 
-  & + label {
-    position: relative;
-    cursor: pointer;
-    padding: 0;
+const getSymbolColor = ({ intermediate, filled, theme }: Props & { theme: Theme }) => {
+  if (!filled) {
+    return theme.utils.getColor('darkGray', 700);
+  } else {
```
Comment (panvourtsis) [inline src/components/CheckBox/CheckBox.style.ts:51]: no need for this else :)

**Review by panvourtsis — APPROVED** [review]

**Comment by panvourtsis** [comment]
LGTM 🎊

**Pull request merged**